### DISCUSSION TITLE
Close company suggestion with escape, but not escape from evil insert?

### DISCUSSION THREAD

**Discussion by rytswd** [company-mode/company-mode#1356] [Q&A] [answered]
Sorry for a newbie question, but I have been searching for a solution for a week or so now, to no avail...

I've got company-mode enabled within my Doom Emacs with evil-mode on. When I want the company-mode to stop suggesting for that particular word / func / etc., I hit escape without thinking, which breaks the typing flow as it goes back to normal mode while also closing company-mode suggestions. Is there some way to close the company suggestion only, preferably with the escape key (due to my muscle memory), but still keep the evil-mode intact?

Let me know if this should be an evil-mode question, I'll send it over there instead 🙏

**Comment by dgutov** [comment, marked as answer]
That's an evil-mode question, please ask it there.

**Reply by rytswd** [reply]
Understood, sorry for the noise!
I'll write back when I find a solution for future references here.

**Reply by rytswd** [reply]
I got a solution for my case, as suggested https://github.com/emacs-evil/evil/issues/1730#issuecomment-1361199359:

```elisp
(define-key company-active-map [escape] 'company-abort)
```

As explained in the above reference, it was that going back to normal-mode would move the cursor and exit company-mode because of that. With this key binding shadowing, it works exactly what I wanted ☺️

**Reply by dgutov** [reply]
Excellent :)
